scripts/hotspot.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+import argparse
+import csv
+import os
+import pysam
+
+"""Function to parse the command line input 
+
+Args:
+    None
+
+Return:
+    Namespace: returns the args as a standard Namespace object
+"""
+def parse_command():
+    parser = argparse.ArgumentParser(description='Flag HotSpots in VCF')
+    parser.add_argument('--hotspots',
+        required=True,
+        help='TSV file containing the hotspots with the following format <csq-field>:<amino-acid[ref]><protein-position[start]> in the 4th column')
+    parser.add_argument('--vcf',
+        required=True,
+        help='Input VCF or VCF.GZ file')
+    parser.add_argument('--csq_field',
+        type=str,
+        required=False,
+        default='SYMBOL',
+        help='CSQ field that matches the format used in the hotspots file')
+    parser.add_argument('--output_basename',
+        type=str,
+        required=False,
+        default="test",
+        help="String to use as basename for output file")
+    args = parser.parse_args()
+    exit = False
+    reasons = []
+    if not os.path.isfile(args.hotspots):
+        exit = True
+        reasons.append("{} is not a file.".format(args.hotspots))
+    if not os.path.isfile(args.vcf):
+        exit = True
+        reasons.append("{} is not a file.".format(args.vcf))
+    if exit:
+        raise FileNotFoundError("{}{}".format(chr(10),chr(10).join(reasons)))
+    if not args.vcf.endswith(".vcf.gz") and not args.vcf.endswith(".vcf"):
+        raise ValueError("{}{} is not a VCF or VCF.GZ".format(chr(10),args.vcf))
+    else:
+        return args
+
+"""Small sanity check for two different feature fields
+
+Args:
+    csq (str): string denoting the CSQ field
+    hotspot (str): a hotspot
+
+Return:
+    str: Info about the CSQ field and hotspot
+"""
+def csq_sanity(field,value):
+    if (field == "Gene" and not value.startswith("ENSG")) or (field == "Feature" and not value.startswith("ENST")):
+        return "CSQ field: {} potentially mismatches hotspot format: {}".format(field,value)
+    else:
+        return "CSQ field: {}; Example hotspot: {}".format(field,value)
+
+"""Function to read the hotspots file and return a list of unique hotspots
+
+Args:
+    infile (file): file containing hotspot information
+    csq (str): csq
+
+Return:
+    list: returns a list of unique hotspot values from the file
+"""
+def process_hotspots(infile,field):
+    hs_list = []
+    with open(infile) as fh:
+        rd = csv.reader(fh, delimiter='\t')
+        for row in rd:
+            if row[0].startswith('#'):
+                continue
+            hs_list.append(row[3].split(','))
+    unique_hs = list(set([items for sublist in hs_list for items in sublist]))
+    print(csq_sanity(field,unique_hs[0]))
```

Comment:
Was this just for debugging or is it supposed to print this every time?

Reply:
It will print once per run. It's something I think will be valuable in the log file for any potential errors that come from people selecting fields that don't match.